azurerm/internal/services/apimanagement/api_management_diagnostic_resource.go

```diff
@@ -256,6 +269,7 @@ func resourceApiManagementDiagnosticRead(d *pluginsdk.ResourceData, meta interfa
 			d.Set("backend_request", nil)
 			d.Set("backend_response", nil)
 		}
+		d.Set("operation_name_format", props.OperationNameFormat)
```

Comment:
which means this could be nil for existing resources - so we'll need to default to that default value here:

```suggestion
		format := "<some default>"
		if props.OperationNameFormat != nil {
			format = string(*props.OperationNameFormat)
		}
		d.Set("operation_name_format", format)
```